backend/pkg/auth/auth.go

```diff
@@ -271,3 +274,175 @@ func RefreshAndCacheNewToken(ctx context.Context, oidcAuthConfig *kubeconfig.Oid
+			return
+		}
+
+		if expiry, err := GetExpiryUnixTimeUTC(claims); err != nil || time.Now().After(expiry) {
+			http.Error(w, "token expired", http.StatusUnauthorized)
+			return
+		}
+
+		username := stringValueFromJMESPaths(claims, usernamePaths)
+		email := stringValueFromJMESPaths(claims, emailPaths)
+		groups := stringSliceFromJMESPaths(claims, groupsPaths)
+
+		writeMeResponse(w, username, email, groups)
+	}
+}
+
+// resolveMePaths applies defaults to the JMESPath expressions used to extract identity claims.
+func resolveMePaths(opts MeHandlerOptions) (string, string, string) {
+	usernamePaths := strings.TrimSpace(opts.UsernamePaths)
+	if usernamePaths == "" {
+		usernamePaths = cfg.DefaultMeUsernamePath
+	}
+
+	emailPaths := strings.TrimSpace(opts.EmailPaths)
+	if emailPaths == "" {
+		emailPaths = cfg.DefaultMeEmailPath
+	}
+
+	groupsPaths := strings.TrimSpace(opts.GroupsPaths)
+	if groupsPaths == "" {
+		groupsPaths = cfg.DefaultMeGroupsPath
+	}
+
+	return usernamePaths, emailPaths, groupsPaths
+}
+
+// readClaimsFromCookie extracts the JWT claims for the given cluster from request cookies.
+func readClaimsFromCookie(r *http.Request, cluster string) (map[string]interface{}, int, string) {
+	token, err := GetTokenFromCookie(r, cluster)
+	if err != nil || token == "" {
+		return nil, http.StatusUnauthorized, "unauthorized"
+	}
+
+	parts := strings.SplitN(token, ".", 3)
+	if len(parts) != 3 || parts[1] == "" {
+		return nil, http.StatusUnauthorized, "invalid token"
+	}
+
+	claims, err := DecodeBase64JSON(parts[1])
+	if err != nil {
+		return nil, http.StatusUnauthorized, "invalid token claims"
+	}
+
+	return claims, 0, ""
+}
+
+// writeMeResponse serializes the identity payload with the standard cache-busting headers.
+func writeMeResponse(w http.ResponseWriter, username, email string, groups []string) {
+	w.Header().Set("Content-Type", "application/json")
+	w.Header().Set("Cache-Control", "no-store, no-cache, must-revalidate, private")
+	w.Header().Set("Pragma", "no-cache")
+	w.Header().Set("Expires", "0")
+	w.Header().Set("Vary", "Cookie")
+	w.Header().Del("ETag")
+	_ = json.NewEncoder(w).Encode(map[string]interface{}{
+		"username": username,
+		"email":    email,
+		"groups":   groups,
+	})
+}
+
+// stringValueFromJMESPaths tries multiple comma-separated JMESPath expressions and returns the first string result.
+func stringValueFromJMESPaths(payload map[string]interface{}, pathCSV string) string {
+	for _, p := range strings.Split(pathCSV, ",") {
+		p = strings.TrimSpace(p)
+		if p == "" {
+			continue
+		}
+
+		res, err := jmespath.Search(p, payload)
+		if err != nil || res == nil {
+			continue
+		}
+
+		switch v := res.(type) {
+		case string:
+			if v != "" {
+				return v
+			}
+		case fmt.Stringer:
+			vs := v.String()
+			if vs != "" {
+				return vs
+			}
+		case float64:
+			return fmt.Sprintf("%v", v)
+		case int64:
+			return fmt.Sprintf("%v", v)
+		case map[string]interface{}:
+			b, _ := json.Marshal(v)
```

Comment:
This was mentioned in another convo that's resolved now, but this error is silently ignored, and I don't think it should be.